pyiceberg/table/snapshots.py

```diff
@@ -429,3 +429,13 @@ def ancestors_of(current_snapshot: Optional[Snapshot], table_metadata: TableMeta
         if snapshot.parent_snapshot_id is None:
             break
         snapshot = table_metadata.snapshot_by_id(snapshot.parent_snapshot_id)
+
+
+def ancestors_between(
+    current_snapshot: Optional[Snapshot], oldest_snapshot: Snapshot, table_metadata: TableMetadata
```

Comment:
This doesn't looks correct: the `current_snapshot` is a required field `oldest_snapshot` could be None and in that case, we also should provide the entire list of ancestors of the `current_snapshot`

I think the implementation that's in progress on this PR is closer to what we'd want: https://github.com/apache/iceberg-python/pull/533/files

Reply:
this should now be resolved, thank you for the link to that